modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystem.java

```diff
@@ -44,29 +45,69 @@ public int getMaxNumericPrecision() {
         return Short.MAX_VALUE;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public int getMaxPrecision(SqlTypeName typeName) {
+        switch (typeName) {
+            case TIME:
+            case TIME_WITH_LOCAL_TIME_ZONE:
+            case TIMESTAMP:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                return TemporalColumnType.MAX_TIME_PRECISION;
+            default:
+                return super.getMaxPrecision(typeName);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int getDefaultPrecision(SqlTypeName typeName) {
+        switch (typeName) {
+            case TIMESTAMP: // DATETIME
+                return TemporalColumnType.DEFAULT_TIME_PRECISION;
```

Comment:
Could you please explain why have you decided to use _<time precision>_ here? In my opinion, we should use _<timestamp precision>_ for both TIMESTAMP and TIMESTAMP W/ TIMEZONE types